rero_ils/modules/serializers/mixins.py

```diff
@@ -94,30 +94,59 @@ def _postprocess_search_aggregations(self, aggregations: dict) -> None:
 class CachedDataSerializerMixin:
     """Class to load and cached resources for serialization process."""
 
-    _resources = {}
+    def __init__(self, limit=2000):
+        """Init.
+
+        :param limit: the maximum number of element to keep in cache.
+        :raises AssertionError: limit is lower or equal to 0.
+        """
+        # TODO :: Use variable from config but this will conflict all tests
+        #    because appcontext is missing
+        assert limit > 0
+        self._resources = {}
+        self._limit = limit
+
+    def append(self, key, resource):
+        """Append a resource into the cache.
+
+        If the cache reached the limit, then the first older element of the
+        cache will be removed to free space to this resource.
+
+        :param key: the resource key.
+        :param resource: the resource to add to the cache.
+        """
+        if len(self._resources) >= self._limit:
+            # remove the first element from dict. Not sure that this is the
+            # order element of the dictionary, but it should.
+            (k := next(iter(self._resources)), self._resources.pop(k))
+        self._resources[key] = resource
 
     @staticmethod
-    def _get_class_key(loader):
+    def _get_key(loader, pid):
         """Get the key where store the resource.
 
         :param loader: Class use to retrieve the resource records.
         :return: The key where store the resource into the cache.
         """
-        return loader if inspect.isclass(loader) else loader.__class__
+        if not inspect.isclass(loader):
+            loader = loader.__class__
```

Comment:
it's not an hack, it's because we can use "Resource" or "ResourcesSearch()
```
get_resource(Document, 'doc1')  # get_record_by_pid from DB
get_resource(DocumentSearch(), 'doc1')  # load record from ES
```
In first case, I get a class, in other case, I get a class instance.